### PR TITLE
terraform automation

### LINKED ISSUE #10: Terraform Automation
**Objectives:**
- Must use 3Musketeers pattern
- Should have at least following targets:
- plan:
	- Run terraform plan and generate an output file to be used by the apply command
- apply:
	- Run the terraform apply command using the output generated by the plan command
- State file can be managed locally as you'll be deploying the same code into each individual account during the development

**Acceptance:**
- make plan and make apply is deploying the entire terraform stack into the account
- there is no credentials in the code
- A README file is updated with instructions on how to run the automation

### PR DESCRIPTION
**Pull request by zarajoy**
updated Makefile and docker-compose.yaml to allow for terraform automation.

make all = init, plan (output plan), apply the plan.

make tf-ci-remove = destroy created infrastructure.

### CODE REVIEW

**Review by mfreitassm — DISMISSED** [review]

File: terraform/Makefile [hunk 1]
```diff
@@ -0,0 +1,43 @@
+##TASKS
+init:
+	bash -x ../scripts/terraform-auto/init.sh
```
Comment (alanlima) [inline terraform/Makefile]: We should use 3M pattern here so the terraform commands should be executed from a docker container.
The reason is that the terraform cli might not be available in the github actions and also to ensure that the same terraform CLI version will be used to execute the commands.

Sample code:

```bash
DOCKER_RUNNER?=docker-compose run --rm

tf-ci-plan:
	@$(DOCKER_RUNNER) ci-terraform init
	@$(DOCKER_RUNNER) ci-terraform plan
.PHONY:tf-ci-plan
```

you will also need to add the `ci-terraform` service in the docker-compose.yml:

```yaml
  ci-terraform:
    image: hashicorp/terraform:0.12.29
    working_dir: /terraform
    volumes:
      - ${PWD}/terraform:/terraform
    environment:
      - AWS_ACCESS_KEY_ID
      - AWS_SECRET_ACCESS_KEY
      - TF_IN_AUTOMATION=yes
```

File: terraform/Makefile [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	bash -x ../scripts/terraform-auto/wipe.sh
+.PHONY: wipe
+
+
+all:
+	make init
+	make plan
+	make apply
+
+.PHONY: all
```
Comment (alanlima) [inline terraform/Makefile]: Use make recipe deps:

```Makefile
alll: init plan apply
.PHONY: all
```

**Review by alanlima — CHANGES_REQUESTED** [review]
Well done @zarajoy 
Just a few adjustments.
Also, to avoid having multiple makefile around the project, could you please add the commands in the root makefile?

File: terraform/Makefile (hunk 1 above)
Comment (zarajoy) [reply terraform/Makefile]: done

**Review by alanlima — DISMISSED** [review]
Good job @zarajoy 
Nice work applying 3M pattern.

**Review by alanlima — APPROVED** [review]

**Pull request merged**